recipe/build.sh

```diff
@@ -1,9 +1,10 @@
 #!/bin/bash
 
-$PYTHON setup.py install --single-version-externally-managed --record record.txt
+# use the pip source to install itself
+PYTHONPATH="./src" $PYTHON -m pip install --no-deps --no-build-isolation . -vv
```

Comment:
Neat!

Reply:
how does that work? the pip program is loaded into memory then the version in memory installs the version on disc?

Reply:
does that remove the need to set `CONDA_ADD_PIP_AS_PYTHON_DEPENDENCY=0` and/or have python as a host dependency?

Reply:
>how does that work? the pip program is loaded into memory then the version in memory installs the version on disc?

Well, everything is always loaded in memory... What it does is just rely on the non-installed (so in-tree) pip to install itself. `-m` will call `src/pip/__main__.py` which is the CLI entrypoint.

Reply:
When importing packages, Python will search and load modules from various location of disk, such as from the `site-packages` directory. The `PYTHONPATH` variable can be use to add locations to the search path.

Reply:
> does that remove the need to set `CONDA_ADD_PIP_AS_PYTHON_DEPENDENCY=0`?

No, `CONDA_ADD_PIP_AS_PYTHON_DEPENDENCY` prevents conda from installing the `pip` conda package when python is listed as a requirement in the recipe. The environment variable is the same as setting `add_pip_as_python_dependency: false` in .condarc.